ui/apps/pmm-compat/src/compat.ts

```diff
@@ -108,10 +102,63 @@ export const initialize = () => {
       messenger.sendMessage({
         id: msg.id,
         type: msg.type,
-        payload: {
-          url: url,
-        },
+        payload: { url },
       });
     },
   });
 };
+
+/**
+ * Wires Grafana theme changes (ThemeChangedEvent) to Percona CSS scheme.
+ * Ensures <html> has the attribute our CSS reads and informs the parent frame.
+ */
+function setupThemeWiring() {
+  // Resolve parent origin robustly (dev vs prod)
+  const getParentOrigin = (): string => {
+    try {
+      const u = new URL(document.referrer);
+      return `${u.protocol}//${u.host}`;
+    } catch {
+      return '*';
+    }
+  };
+
+  const isDev =
+    location.hostname === 'localhost' ||
+    location.hostname === '127.0.0.1' ||
+    /^\d+\.\d+\.\d+\.\d+$/.test(location.hostname);
+
+  const targetOrigin = isDev ? '*' : getParentOrigin();
+
+  // Helper to apply our scheme attribute and notify parent
+  const apply = (incoming: 'light' | 'dark' | string) => {
+    // normalize to 'light' | 'dark'
+    const mode: 'light' | 'dark' = String(incoming).toLowerCase() === 'dark' ? 'dark' : 'light';
+
+    const html = document.documentElement;
+    const scheme = mode === 'dark' ? 'percona-dark' : 'percona-light';
+
+    // Set attributes our CSS reads
+    html.setAttribute('data-md-color-scheme', scheme);
+    html.setAttribute('data-theme', mode);
+    (html.style as CSSStyleDeclaration & { colorScheme: string }).colorScheme = mode;
+    // Notify outer PMM UI (new nav) to sync immediately
+    try {
+      const target = window.top && window.top !== window ? window.top : window.parent || window;
+      target?.postMessage({ type: 'grafana.theme.changed', payload: { mode } }, targetOrigin);
```

Comment:
Use messenger to handle the communication between the `iframe` and `pmm-ui` (no need to handle the target, etc.). We also use predefined message types (see `ui/packages/shared/src/types.ts`)